core/sql/exp/ExpLOBinterface.cpp

```diff
@@ -728,9 +728,9 @@ Lng32 ExpLOBInterfaceUpdate(void * exLobGlob,
   Ex_Lob_Error status;
   Int64 cliError = -1;
   Int64 sourceLen = 0;
-  if((so == Lob_Memory) || (so == Lob_External_File))
+  if(so == Lob_External_File)
     sourceLen = strlen(srcLobData);
-  else if (so == Lob_Buffer)
+  else if ((so == Lob_Buffer) || (so == Lob_Memory))
```

Comment:
For Lob_Memory,  for eg stringtolob() , won't this break since we don't have the target length ?